app/javascript/components/bathroom.jsx

```diff
@@ -20,34 +18,36 @@ const Column = styled.div`
   margin-top: 100px;
 `;
 
-export const Bathroom = ({ cityName }) => (
-  <Query query={getBathroomCode} variables={{ cityName }}>
-    {({ loading, error, data }) => {
-      if (loading) {
-        return <LoadingMessage />;
-      }
-
-      if (error) {
-        return <ErrorMessage />;
-      }
-
-      const { bathroomCode } = data.location;
-      if (!bathroomCode) {
-        return null;
-      }
+export const Bathroom = ({ loading, error, location }) => {
```

Comment:
How does passing in these props instead of the previous one (cityName) reduce side panel queries?

Reply:
The components originally had `<Query>` components, each running a separate query. This switches to fragments that are exported and composed within parent queries, eventually reaching the parent location query in widget-controller. These props are passed down from widget-controller through the children components to as a result of the fragment resolving in the main query.